### DISCUSSION TITLE
Can WatsonWebServer supportwebsocket?

### DISCUSSION THREAD

**Discussion by huangfuhao** [dotnet/WatsonWebserver#120] [Ideas]
Upgrade http/https to ws/wss service

**Comment by jchristn** [comment]
No it cannot.  I'd be open to a PR if you'd like to integrate this capability.

**Comment by joreg** [comment]
I'm also interested in this. I found this for a general explanation of how this works: https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism but do you have an idea of how this could technically work here? would we have a dependency on WatsonWebSocket then? and is it the case that HTTP is completely dead once the websocket is running or do http and websocket then run in parallel?
